src/Miki/Modules/Accounts/AccountsModule.cs

```diff
@@ -1140,39 +1141,46 @@ await builder.ToEmbed()
                 return;
             }
 
-            int streak = 0;
-            string redisKey = $"user:{e.GetAuthor().Id}:daily";
+            ICacheClient cache = e.GetService<ICacheClient>();
+            string redisKey = $"user:{user.Id}:daily";
 
-            var cache = e.GetService<ICacheClient>();
+            if (await cache.ExistsAsync(redisKey).ConfigureAwait(false))
+            {
+                dailyStreak.CurrentStreak = await cache.GetAsync<int>(redisKey).ConfigureAwait(false);
+                await cache.RemoveAsync(redisKey);
+            }
 
-            if(await cache.ExistsAsync(redisKey))
+            if (DateTime.UtcNow <= dailyStreak.LastStreakTime.AddHours(48))
+            {
+                dailyStreak.CurrentStreak++;
+                dailyStreak.LastStreakTime = DateTime.UtcNow;
+            } else
             {
-                streak = await cache.GetAsync<int>(redisKey);
-                streak++;
+                dailyStreak.CurrentStreak = 0;
+                dailyStreak.LastStreakTime = DateTime.UtcNow;
             }
 
-            int amount = dailyAmount + (dailyStreakAmount * Math.Min(100, streak));
+            int finalAmount = (dailyAmount * donatorMultiplier) + ((dailyStreakAmount * donatorMultiplier) * Math.Min(100, (int)dailyStreak.CurrentStreak));
```

Comment:
This seems to be a very long and complex line of code, might be useful to kind of move it to the service layer instead.